mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -302,22 +302,19 @@ class KMeans @Since("1.5.0") (
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
-    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
-    fit(dataset, handlePersistence)
-  }
-
-  @Since("2.2.0")
-  protected def fit(dataset: Dataset[_], handlePersistence: Boolean): KMeansModel = {
```

Comment:
There's another jira focusing on that issue. SPARK-18608. Feel free to share your suggestions in the jira discussion.